source/modulo_core/include/modulo_core/communication/SubscriptionHandler.hpp

```diff
@@ -94,6 +105,27 @@ void SubscriptionHandler<MsgT>::set_subscription(const std::shared_ptr<rclcpp::S
   this->subscription_ = subscription;
 }
 
+template<typename MsgT>
+inline std::function<void(const std::shared_ptr<MsgT>)> SubscriptionHandler<MsgT>::get_callback() {
+  if constexpr (TranslatedT<MsgT> || std::same_as<MsgT, EncodedState>) {
+    return get_translated_callback();
+  } else {
+    return get_raw_callback();
+  }
```

Comment:
just as a mental check for myself that this if / else covers all the cases, it's basically saying that every subscription made with a message type that is not one of the translatable concept types should be handled as a "raw" message. Our definition of "raw message" is when the message on the wire and the message object in memory have the same structure, and thereby we just do a direct assignment  (read_raw_message is `*this->data_ = message`)

As per my other comment, if these are the only two cases, then EncodedState should be part of that translatable message concept.